documentation/docs/specs/revocation_bitmap_2022.md

```diff
@@ -33,7 +33,7 @@ For an issuer to enable verifiers to check the status of a verifiable credential
 
 | Property | Description |
 | :--- | :--- |
-| `id` | The constraints on the `id` property are listed in the [Verifiable Credentials Data Model specification](https://www.w3.org/TR/vc-data-model/). The `id` MUST be a [DID URL](https://www.w3.org/TR/did-core/#did-url-syntax) that resolves to a [Revocation Bitmap Service](#revocation-bitmap-service) in the DID Document of the issuer. |
+| `id` | The constraints on the `id` property are listed in the [Verifiable Credentials Data Model specification](https://www.w3.org/TR/vc-data-model/). The `id` MUST be a [DID URL](https://www.w3.org/TR/did-core/#did-url-syntax) that is the URL to a [Revocation Bitmap Service](#revocation-bitmap-service) in the DID Document of the issuer. It SHOULD include an `index` query set to the same value as `revocationBitmapIndex`, to uniquely identify the `credentialStatus`. If the `index` query is present, implementations SHOULD reject statuses where the `index` query's value does not match `revocationBitmapIndex`. |
```

Comment:
Dereferencing a service endpoint according to the DID Resolution specification uses a different approach.

E.g.
https://w3c-ccg.github.io/did-resolution/#example-9

I find this inconsistent and annoying, and I think the way it is now (i.e. `did:example:123?index=5#service`) would still be compatible since [dereferencing a primary DID Url resource](https://w3c-ccg.github.io/did-resolution/#dereferencing-algorithm-primary) only applies if we use the `?service` query parameter? Is there any reason to switch to their approach, which would be a breaking change?

Reply:
> Is there any reason to switch to their approach, which would be a breaking change?

I completely agree that this feels inconsistent. I would be fine with keeping it as-is now, too, but I'm concerned a conformant resolver would run into branch `3)` in dereferencing the primariy DID Url resource, rather than `2)`, because the DID Url now contains a query:

> 3) Otherwise, if the input [DID URL](https://w3c-ccg.github.io/did-resolution/#dfn-did-url) contains a [DID path](https://w3c-ccg.github.io/did-resolution/#dfn-did-path) and/or [DID query](https://w3c-ccg.github.io/did-resolution/#dfn-did-query):

https://w3c-ccg.github.io/did-resolution/#example-3

Or is that not applicable for some reason?

Reply:
It will take branch `3`, so it's up to the universal resolver implementation what to do at that point.

I think it's fine? The entire service should be validated, so getting only the service endpoint seems inapplicable. I don't know why they introduced a special case to ignore the service type etc. and only extract the endpoint.

Reply:
Yeah probably. Will leave it as-is.